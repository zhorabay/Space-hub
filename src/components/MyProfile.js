import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import '../styles/MyProfile.css';

function MyProfile() {
  const { rockets } = useSelector((state) => state.rockets);
  const { missions } = useSelector((state) => state.missions);
  const missionsData = missions.filter((mission) => mission.reserved);
  const rocketsData = rockets.filter((rocks) => rocks.reserved);

  return (
    <section className="myprofile">
      <div className="mymissions">
        <h2>My Missions</h2>
        {missionsData.length === 0 ? (
          <div className="nomissions">
            <p>No missions joined</p>
            <Link className="join-missions-link" to="/missions">Join Missions</Link>
          </div>
        ) : (
          <ul className="list">
            {missionsData.map((mission) => (
              <li key={mission.id} className="list-table">{mission.name}</li>
            ))}
          </ul>
        )}
      </div>

      <div className="myrockets">
        <h2>My Rockets</h2>
        {rocketsData.length === 0 ? (
          <div className="noreservations">
            <p>No reservations made</p>
            <Link className="make-reservetions-link" to="/rockets">Make Reservations</Link>
          </div>
        ) : (
          <ul className="list">
            {rocketsData.map((rocket) => (
              <li key={rocket.id} className="list-table">{rocket.name}</li>
            ))}
          </ul>
        )}
      </div>
    </section>
  );
}

export default MyProfile;
