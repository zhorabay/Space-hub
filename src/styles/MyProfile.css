.myprofile {
  display: flex;
  gap: 20px;
}

.nomissions,
.noreservations {
  font-style: italic;
}

.join-missions-link,
.make-reservetions-link {
  color: rgb(35 101 255);
}

.join-missions-link:hover,
.make-reservetions-link:hover {
  color: rgb(13, 76, 222);
}

.list {
  list-style: none;
  display: flex;
  flex-direction: column;
}

.list-table {
  list-style: none;
  padding: 12px;
  border: 1px solid grey;
}
