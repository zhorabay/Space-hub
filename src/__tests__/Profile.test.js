import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MyProfile from '../components/Profile';

const mockStore = configureStore([]);

describe('Profile', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      missions: {
        missions: [
          {
            name: 'Test Mission',
            description: 'Test Description',
            reserved: false,
            id: 'test-id',
          },
          {
            name: 'Test Mission 2',
            description: 'Test Description 2',
            reserved: true,
            id: 'test-id 2',
          },
        ],
        pending: false,
        error: false,
      },
      rockets: {
        rockets: [
          {
            name: 'Test Rocket 1',
            description: 'Test Description 1',
            reserved: true,
            id: 'test-id 1',
          },
          {
            name: 'Test Rocket 2',
            description: 'Test Description 2',
            reserved: false,
            id: 'test-id 2',
          },
        ],
        pending: false,
        error: false,
      },
    });
  });

  test('MissionCard rendered correctly', () => {
    const myProfile = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(myProfile).toMatchSnapshot();
  });

  test('Only display reserved rockets and missions', () => {
    const { getByText } = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );

    expect(getByText('Test Mission 2')).toBeInTheDocument();
    expect(getByText('Test Rocket 1')).toBeInTheDocument();
  });
});
