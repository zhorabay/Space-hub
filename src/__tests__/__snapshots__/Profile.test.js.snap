// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionCard MissionCard rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="my-profile"
      >
        <div
          class="my-missions"
        >
          <p
            class="missions-headline"
          >
            My Missions
          </p>
          <ul
            class="missions-list"
          >
            <li>
              Test Mission 2
            </li>
          </ul>
        </div>
        <div
          class="my-rockets"
        >
          <p
            class="rockets-headline"
          >
            My Rockets
          </p>
          <ul
            class="rockets-list"
          >
            <li>
              Test Rocket 1
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="my-profile"
    >
      <div
        class="my-missions"
      >
        <p
          class="missions-headline"
        >
          My Missions
        </p>
        <ul
          class="missions-list"
        >
          <li>
            Test Mission 2
          </li>
        </ul>
      </div>
      <div
        class="my-rockets"
      >
        <p
          class="rockets-headline"
        >
          My Rockets
        </p>
        <ul
          class="rockets-list"
        >
          <li>
            Test Rocket 1
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Profile MissionCard rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="my-profile"
      >
        <div
          class="my-missions"
        >
          <p
            class="missions-headline"
          >
            My Missions
          </p>
          <ul
            class="missions-list"
          >
            <li>
              Test Mission 2
            </li>
          </ul>
        </div>
        <div
          class="my-rockets"
        >
          <p
            class="rockets-headline"
          >
            My Rockets
          </p>
          <ul
            class="rockets-list"
          >
            <li>
              Test Rocket 1
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="my-profile"
    >
      <div
        class="my-missions"
      >
        <p
          class="missions-headline"
        >
          My Missions
        </p>
        <ul
          class="missions-list"
        >
          <li>
            Test Mission 2
          </li>
        </ul>
      </div>
      <div
        class="my-rockets"
      >
        <p
          class="rockets-headline"
        >
          My Rockets
        </p>
        <ul
          class="rockets-list"
        >
          <li>
            Test Rocket 1
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
